presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -367,6 +376,24 @@ public void enforceMemoryLimits()
     }
 
     /**
+     * Enforce timeout at the query level
+     */
+    public void enforceQueryMaxRuntimeLimits()
+    {
+        Iterable<QueryExecution> runningQueries = queries.values().stream()
+                .filter(query -> query.getQueryInfo().getState() == RUNNING)
+                .collect(toImmutableList());
+
+        for (QueryExecution runningQuery : runningQueries) {
+            Duration queryMaxRuntime = SystemSessionProperties.getQueryMaxRuntime(runningQuery.getSession());
```

Comment:
@dain In this case, should it be changed to
```
if (query.getQueryInfo().getState().isDone()) {
    continue;
}
```
or should it be removed altogether? I read `QueryStateMachine.transitionToFailed`. Both works, but which is better practice?